### PR TITLE
Add user type visibility options to new search engine

### PR DESCRIPTION
**Pull request by crisptrutski**
Refs https://github.com/metabase/metabase/issues/40964

This matches legacy search's logic. I think it's the very last thing!

### CODE REVIEW

File: src/metabase/models/model_index.clj [hunk 1]
```diff
@@ -187,6 +187,7 @@
 
 (search/define-spec "indexed-entity"
   {:model        :model/ModelIndexValue
+   :visibility   :app-user
```
Comment (crisptrutski) [inline src/metabase/models/model_index.clj:190]: Not sure what to call this 

- regular user?
- actual user?
- non-impersonated-or-sandbox-user?

Comment (qnkhuat) [reply src/metabase/models/model_index.clj:190]: lol this is a hard one, maybe @noahmoss can help here :D

**Review by qnkhuat — APPROVED** [review]
would be nice to remove the `premium-features/impersonated-user?` and `premium-features/sandboxed-user?` reference in search permissions but It's already there so no blocking.

**Comment by crisptrutski** [comment]
> would be nice to remove the `premium-features/impersonated-user?` and `premium-features/sandboxed-user?` reference in search permissions but It's already there so no blocking.

Definitely going to take a stab as part of the clean-up!

**Pull request merged**